astropy/table/index.py

```diff
@@ -817,13 +808,73 @@ def __getitem__(self, item):
             rows = []
             for key in item:
                 rows.extend(index.find((key,)))
+        return rows
+
+    def __getitem__(self, item):
+        '''
+        Retrieve Table rows by value slice.
+
+        Parameters
+        ----------
+        item : column element, list, ndarray, slice or tuple
+            Can be a value of the table primary index, a list/ndarray
+            of such values, or a value slice (both endpoints are included).
+            If a tuple is provided, the first element must be
+            an index to use instead of the primary key, and the
+            second element must be as above.
+        '''
+        rows = self._get_rows(item)
 
         if len(rows) == 0:  # no matches found
             raise KeyError('No matches found for key {0}'.format(item))
         elif len(rows) == 1:  # single row
             return self.table[rows[0]]
         return self.table[rows]
 
+    def __setitem__(self, key, value):
+        '''
+            assign Table row's indices  by value slice.
+
+            Parameters
+                        ----------
+            key : column element, list, ndarray, slice or tuple
+                  Can be a value of the table primary index, a list/ndarray
+                  of such values, or a value slice (both endpoints are included).
+                  If a tuple is provided, the first element must be
+                  an index to use instead of the primary key, and the
+                  second element must be as above.
+
+            value : New values of the row elements.
+                    Can be a list of tuples/lists to update the row.
+        '''
+        rows = self._get_rows(key)
+        if len(rows) == 0:  # no matches found
+            raise KeyError('No matches found for key {0}'.format(key))
+        elif len(rows) == 1:  # single row
+            self.table[rows[0]] = value
+        else:  # multiple rows
+            for i, row in enumerate(rows):
```

Comment:
Check that `rows` and `value` have the same length.  Then use `for row, val in zip(rows, value):  ...`.

Reply:
"Check that rows and value have the same length."!..if not should i throw any exception??

Reply:
Yes.

Reply:
ok!